### PR TITLE
Feature/Add Ownership Change to Listener Interface

### PR DESCRIPTION
**Pull request by Hany-Almnaem**
## Overview
This PR implements the `ownerChanged` hook in the `PDPListener` interface, enabling continued tracking of approved SPs when they update their owner address in `PDPVerifier`. This supports secure key rotation for providers.

### Interface Update
- Added `ownerChanged(uint256 proofSetId, address oldOwner, address newOwner)` to `PDPListener`
- Documentation added for clarity

### PDPVerifier Implementation
- Calls `ownerChanged` after emitting `ProofSetOwnerChanged` in `claimProofSetOwnership`
- Uses `try/catch` to prevent listener revert from breaking main transaction

### Listener Implementations
- **SimplePDPService**: No-op implementation
- **TestingRecordKeeperService**: Full state-tracking and revert testing
- **BadListener**: No-op implementation for compatibility
- **ExtraDataListener**: No-op implementation for compatibility

### Test Coverage
- `testOwnerChangedCalledOnOwnershipTransfer`: Checks params and event
- `testNoListenerNoRevert`: Ensures no revert if listener is zero address
- `testListenerRevertDoesNotRevertMainTx`: Ensures listener revert doesn’t block main flow

## Testing
- All existing and new ownership listener tests pass
- Edge cases are handled

## Next Steps (Planned in PR 2)
- Update `PandoraService` to implement `ownerChanged` and update payment rail mapping
- Bump pdp submodule pointer in filecoin-services
- Add integration tests for the complete ownership flow

## Related
FilOzone/filecoin-services[#15](https://github.com/FilOzone/filecoin-services/issues/15)

### CODE REVIEW

File: test/PDPVerifier.t.sol [hunk 1]
```diff
@@ -704,6 +704,38 @@ contract ProofBuilderHelper is Test {
 // TestingRecordKeeperService is a PDPListener that allows any amount of proof challenges
 // to help with more flexible testing.
 contract TestingRecordKeeperService is PDPListener, PDPRecordKeeper {
+    // State to record the last ownerChanged call
+    uint256 public lastOwnerChangedProofSetId;
```
Comment (ZenGround0) [inline test/PDPVerifier.t.sol:708]: This contract isn't the place to be adding these owner specific things.  Its purpose is to take a record of listener calls and make sure they match.  Please remove these fields, the getter and setter, the code in ownerChanged and make the ownerChanged simply receive proof set event.  Please implement this useful testing functionality in a specific testing mock owner-change listener contract right before the `PDPVerifierOwnershipListenerTest`

File: test/PDPVerifier.t.sol [hunk 2]
```diff
@@ -1804,4 +1840,64 @@ contract PDPVerifierMigrateTest is Test {
...
+        assertEq(listener.lastOwnerChangedNewOwner(), nextOwner, "New owner mismatch");
+        assertEq(listener.ownerChangedCallCount(), 1, "ownerChanged should be called once");
+    }
+
+    function testNoListenerNoRevert() public {
+        // Create proof set with no listener
+        uint256 setId = pdpVerifier.createProofSet{value: PDPFees.sybilFee()}(address(0), empty);
+        pdpVerifier.proposeProofSetOwner(setId, nextOwner);
+        vm.prank(nextOwner);
+        // Should not revert even though listenerAddr is zero
```
Comment (ZenGround0) [inline test/PDPVerifier.t.sol:1885]: As established in previous comment this behavior should flip after review response.

**Review by ZenGround0 — CHANGES_REQUESTED** [review]
A few important changes but overall this is in the right direction.

File: .gitmodules [hunk 3]
```diff
@@ -7,3 +7,6 @@
 [submodule "lib/openzeppelin-contracts-upgradeable"]
 	path = lib/openzeppelin-contracts-upgradeable
 	url = https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable
+[submodule "lib/pyth-sdk-solidity"]
```
[the remark this answers is not in the document]
Comment (Hany-Almnaem) [reply .gitmodules:10]: Great, I’ll move all .gitmodules/dependency changes into a separate PR and revert them here

[the remark this answers is not in the document]
Comment (Hany-Almnaem) [reply .gitmodules]: done, please review https://github.com/FilOzone/pdp/pull/172

File: src/PDPVerifier.sol [hunk 4]
```diff
@@ -23,6 +23,8 @@ interface PDPListener {
     // Note: extraData not included as proving messages conceptually always originate from the SP
     function possessionProven(uint256 proofSetId, uint256 challengedLeafCount, uint256 seed, uint256 challengeCount) external;
     function nextProvingPeriod(uint256 proofSetId, uint256 challengeEpoch, uint256 leafCount, bytes calldata extraData) external;
+    /// @notice Called when proof set ownership is changed in PDPVerifier.
+    function ownerChanged(uint256 proofSetId, address oldOwner, address newOwner) external;
```
[the remark this answers is not in the document]
Comment (Hany-Almnaem) [reply src/PDPVerifier.sol]: Done. The ownerChanged function in PDPListener and all implementing contracts now accept the extraData parameter.

File: src/PDPVerifier.sol [hunk 5]
```diff
@@ -270,6 +272,10 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         proofSetOwner[setId] = msg.sender;
         delete proofSetProposedOwner[setId];
         emit ProofSetOwnerChanged(setId, oldOwner, msg.sender);
+        address listenerAddr = proofSetListener[setId];
+        if (listenerAddr != address(0)) {
+            try PDPListener(listenerAddr).ownerChanged(setId, oldOwner, msg.sender) {} catch {}
```
[the remark this answers is not in the document]
Comment (Hany-Almnaem) [reply src/PDPVerifier.sol]: he try/catch around the listener call in claimProofSetOwnership has been removed, so listener reverts now propagate as intended.

File: test/PDPVerifier.t.sol (hunk 1 above)
Comment (Hany-Almnaem) [reply test/PDPVerifier.t.sol]: All owner-specific fields and logic have been removed from TestingRecordKeeperService.
I introduced  MockOwnerChangedListener contract for ownership-change-related assertions, as you suggest.

File: src/PDPVerifier.sol [hunk 6]
```diff
@@ -274,7 +274,7 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         emit ProofSetOwnerChanged(setId, oldOwner, msg.sender);
         address listenerAddr = proofSetListener[setId];
         if (listenerAddr != address(0)) {
-            try PDPListener(listenerAddr).ownerChanged(setId, oldOwner, msg.sender) {} catch {}
+            PDPListener(listenerAddr).ownerChanged(setId, oldOwner, msg.sender, "");
```
Comment (ZenGround0) [inline src/PDPVerifier.sol:277]: Apologies I should have explained this better: please update the interface to claimProofSetOwnership to accept the extraBytes data.  The way this pattern works is that the caller can provide arbitrary data to be threaded through the listener call.  You'll need to add a `bytes` param to `claimProofSetOwnership` and then pass it in here instead of this empty string.

File: test/PDPVerifier.t.sol [hunk 7]
```diff
@@ -1529,6 +1499,7 @@ contract PDPListenerIntegrationTest is Test {
     BadListener badListener;
     uint256 constant challengeFinalityDelay = 2;
     bytes empty = new bytes(0);
+    bytes testExtraData = "test extra data";
```
Comment (ZenGround0) [inline test/PDPVerifier.t.sol:1502]: I don't understand the point of this field instead of empty -- we're not testing the extra data handling of the listener only its error propagation.

File: test/PDPVerifier.t.sol [hunk 8]
```diff
@@ -1542,63 +1513,68 @@ contract PDPListenerIntegrationTest is Test {
     }
 
     function testListenerPropagatesErrors() public {
-
-        // Can't create a proof set with a bad listener
+        // CREATE operation
```
Comment (ZenGround0) [inline test/PDPVerifier.t.sol:1516]: The new comments are not adding enough clarity to be worth keeping.  `OperationType.CREATE` makes it clear which operation we're dealing with from reading the code.  But removing the old comment is fine -- it also wasn't adding enough clarity to be worth it.

**Review by ZenGround0 — COMMENTED** [review]
Almost good to go, just one serious change to address.

**Comment by ZenGround0** [comment]
Thank you @Hany-Almnaem
